### PR TITLE
Refactored angular services for tweetfeed

### LINKED ISSUE #393: Refactor angular services in front-end apps
The front-end apps mainly use $http or $resource for ajax calls, or using $scope.search from /html/js/loklak.js. Refactoring it to a separate service file will help contributors better organize their app, as the apps may not use the same $scope variables.

### PR DESCRIPTION
**Pull request by leonmak**
To fix #393, so that services can be more reusable.

### CODE REVIEW

File: html/apps/tweetfeed/services/dataService.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+(function(){
+    angular.module('myApp')
+    
+    .factory('dataService',
```
Comment (sudheesh001) [inline html/apps/tweetfeed/services/dataService.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2614812995/issues/source?bid=3288751&fileBranchId=3312487#l4)

File: html/apps/tweetfeed/services/dataService.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            } else {
+                uri = defaultUri;
+            }
+            
+            var promise = $http.get(uri)
+            .then(searchCompleted)
+            .catch(searchFailed);
+            
+            function searchCompleted(response){
+                if (typeof response.data === 'object') {
```
Comment (sudheesh001) [inline html/apps/tweetfeed/services/dataService.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2614812995/issues/source?bid=3288751&fileBranchId=3312487#l28)

File: html/apps/tweetfeed/services/dataService.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            var uri = "";
+            if(queryType === "search"){
+                console.log(queryTerm);
+                uri = "/api/search.json?q=" + queryTerm;
+            } else {
+                uri = defaultUri;
+            }
+            
+            var promise = $http.get(uri)
+            .then(searchCompleted)
```
Comment (sudheesh001) [inline html/apps/tweetfeed/services/dataService.js]: Codacy Issue found: ["searchCompleted" was used before it was defined](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2614812995/issues/source?bid=3288751&fileBranchId=3312487#l24)

File: html/apps/tweetfeed/services/dataService.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            .catch(searchFailed);
+            
+            function searchCompleted(response){
+                if (typeof response.data === 'object') {
+                    return response.data.statuses;
+                } else {
+                    // invalid response
+                    console.log("Failed");
```
Comment (sudheesh001) [inline html/apps/tweetfeed/services/dataService.js]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2614812995/issues/source?bid=3288751&fileBranchId=3312487#l32)

File: html/apps/tweetfeed/services/dataService.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            getTweets: getTweets
+        }
+        
+        function getTweets(queryType, queryTerm){
+            
+            // user's tweets latest first
+            var defaultUri = "/api/search.json?q=loklak";
+            var uri = "";
+            if(queryType === "search"){
+                console.log(queryTerm);
```
Comment (sudheesh001) [inline html/apps/tweetfeed/services/dataService.js]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2614812995/issues/source?bid=3288751&fileBranchId=3312487#l17)

File: html/apps/tweetfeed/services/dataService.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                uri = "/api/search.json?q=" + queryTerm;
+            } else {
+                uri = defaultUri;
+            }
+            
+            var promise = $http.get(uri)
+            .then(searchCompleted)
+            .catch(searchFailed);
```
Comment (sudheesh001) [inline html/apps/tweetfeed/services/dataService.js]: Codacy Issue found: ["searchFailed" was used before it was defined](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2614812995/issues/source?bid=3288751&fileBranchId=3312487#l25)

File: html/apps/tweetfeed/services/dataService.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            }
+            
+            function searchFailed(error) {
+                console.log(error);
+            }
+            
+            return promise;
+        }
+    }
+)
```
Comment (sudheesh001) [inline html/apps/tweetfeed/services/dataService.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2614812995/issues/source?bid=3288751&fileBranchId=3312487#l43)

File: html/apps/tweetfeed/services/dataService.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
+(function(){
+    angular.module('myApp')
```
Comment (sudheesh001) [inline html/apps/tweetfeed/services/dataService.js]: Codacy Issue found: ["angular" is not defined.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2614812995/issues/source?bid=3288751&fileBranchId=3312487#l2)

File: html/apps/tweetfeed/services/dataService.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    
+    .factory('dataService', 
+    function($http){
+        
+        return {
+            getTweets: getTweets
```
Comment (sudheesh001) [inline html/apps/tweetfeed/services/dataService.js]: Codacy Issue found: ["getTweets" was used before it was defined](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2614812995/issues/source?bid=3288751&fileBranchId=3312487#l8)

File: html/apps/tweetfeed/services/dataService.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+                if (typeof response.data === 'object') {
+                    return response.data.statuses;
+                } else {
+                    // invalid response
+                    console.log("Failed");
+                }
+            }
+            
+            function searchFailed(error) {
+                console.log(error);
```
Comment (sudheesh001) [inline html/apps/tweetfeed/services/dataService.js]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2614812995/issues/source?bid=3288751&fileBranchId=3312487#l37)

**Comment by leonmak** [comment]
Merging after review

**Pull request merged**